### PR TITLE
tests: Add a couple negative tests for block decoding

### PR DESCRIPTION
**Pull request by rvl**
Relates to #11.
Add some tests of decoding non-block data.

### CODE REVIEW

File: test/unit/Cardano/Wallet/BinarySpec.hs [hunk 1]
```diff
@@ -61,6 +63,17 @@ spec = do
             let decoded = unsafeDeserialiseFromBytes decodeBlock bs
             decoded `shouldBe` block3
 
+        it "should fail to decode a junk block" $ do
+            let junk = mconcat (replicate 100 "junk")
+                decoded = CBOR.deserialiseFromBytes decodeBlock junk
+            decoded `shouldSatisfy` isLeft
+
+        it "should fail to decode a block with block header data" $ do
+            bs <- L8.readFile "test/data/Cardano/Wallet/BinarySpec-block-header-1"
```
Comment (jonathanknowles) [inline test/unit/Cardano/Wallet/BinarySpec.hs]: https://github.com/input-output-hk/cardano-wallet/wiki/Coding-Standards#limit-line-length-to-80-characters

**Review by jonathanknowles — DISMISSED** [review]
Looks good to me, though still need to bring code in line with https://github.com/input-output-hk/cardano-wallet/wiki/Coding-Standards

File: test/unit/Cardano/Wallet/BinarySpec.hs (hunk 1 above)
Comment (KtorZ) [reply test/unit/Cardano/Wallet/BinarySpec.hs]: Fixed :+1:

**Comment by KtorZ** [comment]
@piotr-iohk I've fixed @jonathanknowles remarks & added a test to unserialize a block contains MANY (many = 2) transactions.

**Review by piotr-iohk — APPROVED** [review]
lgtm :+1:

**Pull request merged**